main.js

```diff
@@ -0,0 +1,209 @@
+const toolsBtn = document.querySelectorAll('.tool');
+const canvas = document.querySelector('canvas');
+const colorFill = document.querySelector('#fill-color');
+const sizeSlider = document.querySelector('#size-slider');
+const colorBtns = document.querySelectorAll('.color .option');
+const colorPicker = document.querySelector('#color-picker');
+const clearCanvas = document.querySelector('.clear-board');
+const saveCanvas = document.querySelector('.save-board');
+const redoCanvas = document.querySelector('.redo-board');
+const undoCanvas = document.querySelector('.undo-board');
+
+const ctx = canvas.getContext("2d");
+
+let selectedTool = "brush",
+ brushWidth = 5,
+ selectedColor = "black",
+ isDrawing = false,
+ prevMouseX,
+ prevMouseY,
+ snapShot,
+ undoStack=[],
+ redoStack = [];
+
+const setCanvasBackground = () => {
+    ctx.fillStyle = '#fff';
+    ctx.fillRect(0, 0, canvas.width, canvas.height);
+    ctx.fillStyle = selectedColor;
+}
+
+window.addEventListener("load", () => {
+    canvas.width = canvas.offsetWidth;
+    canvas.height = canvas.offsetHeight;
+    canvas.style.cursor = "crosshair";
+    setCanvasBackground();
+})
+
+const drawRect = (e) => {
+    if (!colorFill.checked) {
+        ctx.strokeRect(e.offsetX, e.offsetY, prevMouseX - e.offsetX, prevMouseY = e.offsetY);
```

Comment:
Dreptunghiul se desenează ciudat. Mai degrabă ar trebui să fie:

```suggestion
        ctx.strokeRect(prevMouseX, prevMouseY, e.offsetX - prevMouseX, e.offsetY - prevMouseY);
```